streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicConfig.java

```diff
@@ -109,7 +90,11 @@ void setNumberOfPartitions(final int numberOfPartitions) {
     }
 
     void setRetentionMs(final long retentionMs) {
-        if (!logConfig.containsKey(InternalTopicManager.RETENTION_MS)) {
+        if (topicType != InternalTopicType.WINDOWED_STORE_CHANGELOG) {
+            throw new IllegalStateException("Setting retention.ms called for not windowed store changelog topics: " + this.toString());
+        }
+
+        if (!topicConfigs.containsKey(TopicConfig.RETENTION_MS_CONFIG)) {
```

Comment:
Not part of this PR: but why is `topicConfig` preferred over this method call?

Reply:
Because user-customized values are always respected over library defaults, in this case the `window retention + additional retention` is the library defaults, and hence should yield to user specified values.